### ISSUE TITLE
show_help_message() doesn’t check if headers were already sent

### ISSUE THREAD

**Issue by ausi** [contao/core#8488]
If an error occurs after the page was sent to the browser (e.g. in `FrontendTemplate::addToSearchIndex()` or in a `postFlushData` hook) the error handler calls `show_help_message()` which adds the error template to the already sent output. Can be reproduced by adding this to the *initconfig.php*:

```php
$GLOBALS['TL_HOOKS']['postFlushData'][] = ['MyHooks', 'postFlushData'];
class MyHooks {
	public function postFlushData($buffer, $template) {
		throw new Exception;
	}
}
```

IMO this should be fixed by changing [system/helper/functions.php:125](https://github.com/contao/core/blob/549738f9a41974d963bd09a3a83ae0f46f887f01/system/helper/functions.php#L125) to:

```php
if (ini_get('display_errors') || headers_sent())
```

**Comment by leofeyer** [comment]
I see the point but how does the change fix it? The error message will still not be shown, will it?

**Comment by ausi** [comment]
> The error message will still not be shown, will it?

The error should not be shown because the page was already sent with a `200` status code. The problem in the current version is, that the error gets printed additionally to the rest of the page.

**Comment by leofeyer** [comment]
Fixed in 43be3a17db026384dfcdabf0cba117e618bc603b.
